### PR TITLE
js/connection: Use jest for tests

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

Use `jest` and `@testing-library` for tests instead of
`js-test-runner`/`mocha`/`chai`/`sinon`/`enzyme`.

Also set up linting of the tests.

I also wound up having to fix an include loop in the state files, and
declaring that ./state/store.jsx has side effects.

#### Jetpack product discussion

p9dueE-5j2-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Is CI happy?
* Are the tests still being run?
* Do the changes to the tests make sense?

### CODE REVIEW

**Review by coder-karen — DISMISSED** [review]
Looks good, tests tun well.

**Comment by anomiex** [comment]
Needs re-approval after merge

**Review by coder-karen — APPROVED** [review]

**Comment by anomiex** [comment]
r247754-wpcom

**Pull request merged**
